deployer/hub.py

```diff
@@ -41,6 +42,39 @@ def auth(self):
         else:
             raise ValueError(f'Provider {self.spec["provider"]} not supported')
 
+    def ensure_docker_credhelpers(self):
+        """
+        Setup credHelper for current hub's image registry.
+
+        Most image registries (like ECR, GCP Artifact registry, etc) use
+        a docker credHelper (https://docs.docker.com/engine/reference/commandline/login/#credential-helpers)
+        to authenticate, rather than a username & password. This requires an
+        entry per registry in ~/.docker/config.json.
+
+        This method ensures the appropriate credential helper is present
+        """
+        image_name = self.spec['image_repo']
+        registry = image_name.split('/')[0]
+
+        helper = None
+        # pkg.dev is used by Google Cloud Artifact registry
+        if registry.endswith('pkg.dev'):
+            helper = 'gcloud'
+
+        if helper is not None:
+            dockercfg_path = os.path.expanduser('~/.docker/config.json')
+            try:
+                with open(dockercfg_path) as f:
+                    config = json.load(f)
+            except FileNotFoundError:
+                config = {}
+
+            helpers = config.get('credHelpers', {})
+            helpers[registry] = helper
```

Comment:
I think we should add a check here to see if there is already the expected pair of `registry-helper`.
Otherwise, I think we will end up modifying the config file every time we build the image.

But since building the image already takes a loooooooot of time, the time to update the config file doesn't even matter, so my suggestion shouldn't block this PR  that already looks LGTM :rocket:

Reply:
@GeorgianaElena good catch! Fixed :)